src/crawlee/storages/_key_value_store.py

```diff
@@ -175,3 +182,56 @@ async def get_public_url(self, key: str) -> str:
             The public URL for the given key.
         """
         return await self._resource_client.get_public_url(key)
+
+    async def get_auto_saved_value(self, key: str, default_value: dict[str, Any] | None = None) -> dict[str, Any]:
+        """Gets a value from store that will be automatically saved on changes.
+
+        Args:
+            key: Key of the record, to store the value.
+            default_value: value to be used if the record does not exist yet. Should be a dictionary
+
+        Returns:
+            Returns the value of the key
+        """
+        default_value = {} if default_value is None else default_value
+
+        if key in self._cache:
+            return self._cache[key]
+
+        value = await self.get_value(key, default_value)
+
+        self._cache[key] = value
+
+        self._ensure_persist_event()
+
+        return cast(dict[str, Any], value)
```

Comment:
I believe the cast is not necessary, correct?

The returned value of `get_value` is the same type of `default_value`, which is `dict[str, Any]`.